.jenkins/cache.sh

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env bash
+
+set -e
+
+cmd=$1
+key=$2
+paths=( "$@" )
+unset paths[0]  # cmd is not a path to be cached
+unset paths[1]  # key is not a path to be cached
+
+if [ "$cmd" != "save" ] && [ "$cmd" != "restore" ]; then
+    echo "cmd must be one of 'save' or 'restore', got $cmd"
+    exit 1
+fi
+
+if [ -z $PACE_CACHE_DIR ]; then
+    cache_dir=~/.cache/pace
+    mkdir -p $cache_dir
+else
+    cache_dir=$PACE_CACHE_DIR
+fi
+
+key_cache=$cache_dir/$key
+target_dir=$(pwd)
+
+
+if [ "$cmd" == "save" ]; then
+    if [ -d $key_cache ]; then
+        echo "cache for key $key_cache already exists, skipping cache step"
+    else
+        mkdir -p $key_cache
+        for path in "${paths[@]}"; do
```

Comment:
should there be a warning here if the remainder of paths is of size 0?

Reply:
There should probably be an error. I'll update it to do so.